spectec/spec/7-module.watsup

```diff
@@ -0,0 +1,236 @@
+;; Allocation
+
+def $alloc_import(moduleinst, import*, externval*) : moduleinst
+
+def $alloc_import(m, epsilon, epsilon) = m
+
+def $alloc_import(m, import import'*, externval externval'*) = m_res
+  -- if import = IMPORT name name' externtype
+  -- if externtype = FUNC functype
+  -- if externval = FUNC fa
+  -- if m_new = m[.FUNC =.. fa]
+  -- if m_res = $alloc_import(m_new, import'*, externval'*)
+
+def $alloc_import(m, import import'*, externval externval'*) = m_res
+  -- if import = IMPORT name name' externtype
+  -- if externtype = GLOBAL globaltype
+  -- if externval = GLOBAL ga
+  -- if m_new = m[.GLOBAL =.. ga]
+  -- if m_res = $alloc_import(m_new, import'*, externval'*)
+
+def $alloc_import(m, import import'*, externval externval'*) = m_res
+  -- if import = IMPORT name name' externtype
+  -- if externtype = TABLE tabletype
+  -- if externval = TABLE ta
+  -- if m_new = m[.TABLE =.. ta]
+  -- if m_res = $alloc_import(m_new, import'*, externval'*)
+
+def $alloc_import(m, import import'*, externval externval'*) = m_res
+  -- if import = IMPORT name name' externtype
+  -- if externtype = MEM memtype
+  -- if externval = MEM ma
+  -- if m_new = m[.MEM =.. ma]
+  -- if m_res = $alloc_import(m_new, import'*, externval'*)
+
+
+def $alloc_func(state, func*) : (store, funcaddr*)
```

Comment:
Here and below we are conflating the allocx functions with the allocx* liftings. Could we keep them separate? In particular, for the Latex spec, we will only want to include the allocx ones, and not write out each of the others.